### PR TITLE
Update `parse_multipart` return type to support Python 3.7

### LINKED ISSUE #2473: `cgi.parse_multipart` return type incorrect in Python 3.7
As per https://docs.python.org/3/library/cgi.html#cgi.parse_multipart:

> Changed in version 3.7: Added the encoding and errors parameters. For non-file fields, the value is now a list of strings, not bytes.

However, https://github.com/python/typeshed/blob/master/stdlib/2and3/cgi.pyi includes:

```python
def parse_multipart(fp: IO[Any], pdict: Mapping[str, bytes]) -> Dict[str, List[bytes]]: ...
```

The return type is therefore not correct for Python 3.7.

### PR DESCRIPTION
**Pull request by adamtheturtle**
This is intended to fix #2473.

### CODE REVIEW

**Comment by adamtheturtle** [comment]
Thank you for the review @JelleZijlstra . I have updated this PR.

File: stdlib/2and3/cgi.pyi [hunk 1]
```diff
@@ -7,7 +7,10 @@ def parse(fp: IO[Any] = ..., environ: Mapping[str, str] = ...,
           keep_blank_values: bool = ..., strict_parsing: bool = ...) -> Dict[str, List[str]]: ...
 def parse_qs(qs: str, keep_blank_values: bool = ..., strict_parsing: bool = ...) -> Dict[str, List[str]]: ...
 def parse_qsl(qs: str, keep_blank_values: bool = ..., strict_parsing: bool = ...) -> Dict[str, List[str]]: ...
-def parse_multipart(fp: IO[Any], pdict: Mapping[str, bytes]) -> Dict[str, List[bytes]]: ...
+if sys.version_info >= (3, 7):
+    def parse_multipart(fp: IO[Any], pdict: Mapping[str, bytes]) -> Dict[str, List[Union[str, bytes]]]: ...
```
[the remark this answers is not in the document]
Comment (JelleZijlstra) [reply stdlib/2and3/cgi.pyi]: I think you added them to the wrong branch.

[the remark this answers is not in the document]
Comment (adamtheturtle) [reply stdlib/2and3/cgi.pyi]: So I did!

**Comment by adamtheturtle** [comment]
Thanks! Oops - ready for round three.

File: stdlib/2and3/cgi.pyi [hunk 2]
```diff
@@ -7,7 +7,10 @@ def parse(fp: IO[Any] = ..., environ: Mapping[str, str] = ...,
           keep_blank_values: bool = ..., strict_parsing: bool = ...) -> Dict[str, List[str]]: ...
 def parse_qs(qs: str, keep_blank_values: bool = ..., strict_parsing: bool = ...) -> Dict[str, List[str]]: ...
 def parse_qsl(qs: str, keep_blank_values: bool = ..., strict_parsing: bool = ...) -> Dict[str, List[str]]: ...
-def parse_multipart(fp: IO[Any], pdict: Mapping[str, bytes]) -> Dict[str, List[bytes]]: ...
+if sys.version_info >= (3, 7):
+    def parse_multipart(fp: IO[Any], pdict: Mapping[str, bytes], encoding: str, errors: str) -> Dict[str, List[Union[str, bytes]]]: ...
```
Comment (srittau) [inline stdlib/2and3/cgi.pyi]: `encoding` and `errors` have default arguments, so they should get a `= ...`.

Also, the return value should be `Dict[str, List[Any]]`, unfortunately. `List[Any]` is returned by `FieldStorage.getlist()` (which is called by `parse_multipart()`). Using an `Union` means that each caller has to use a cast or instance check on the return value of `parse_multipart()`.

**Comment by adamtheturtle** [comment]
Thank you both for your patience and rigour.

**Comment by srittau** [comment]
Thanks!

**Pull request merged**